### PR TITLE
Add shell style checks based on ShellCheck

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: cmake/test-targets.cmake [hunk 1]
```diff
@@ -37,6 +37,17 @@ add_test(
     WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
 )
 
+find_program(SHELLCHECK_PATH shellcheck)
+if (SHELLCHECK_PATH)
+    add_test(
+        NAME test-local-shellcheck
+        COMMAND shellcheck -x $(file --mime-type script/* t/* container/worker/*.sh tools/* | sed -n 's/^\(.*\):.*text\/x-shellscript.*$/\\1/p')
```
Comment (Martchus) [inline cmake/test-targets.cmake:44]: SHELLCHECK_PATH should be used there.

File: tools/check-shellcheck [hunk 2]
```diff
@@ -0,0 +1 @@
+shellcheck -x $(file --mime-type tools/* | sed -n 's/^\(.*\):.*text\/x-shellscript.*$/\1/p')
```
Comment (Martchus) [inline tools/check-shellcheck:1]: You've been adding the command directly withing CMake so this looks like a leftover.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: tools/check-yaml-syntax [hunk 3]
```diff
@@ -1,3 +1,3 @@
 #!/bin/sh -e
 # fall back to find if there is no git, e.g. in package builds
-"${1:-yamllint}" --strict $(git ls-files "*.yml" "*.yaml" 2>/dev/null || find -name '*.y*ml')
+"${1:-yamllint}" --strict "$(git ls-files "*.yml" "*.yaml" 2>/dev/null || find . -name '*.y*ml')"
```
Comment (BillAnastasiadis) [inline tools/check-yaml-syntax:3]: Is quoting the command substitution the intended behavior?  Looking at the ci error

`No such file or directory: '.github/stale.yml\n.github/workflows/ci.yml\n.github/workflows/ci_extended.yml\n.hadolint.yaml\n.mergify.yml\ncodecov.yml\ndependencies.yaml'`

it seems that the command needs the splitted `$()` output and not the single string `"$()"` one.

File: cmake/test-targets.cmake (hunk 1 above)
Comment (okurz) [reply cmake/test-targets.cmake]: I needed to change that now similar as we do for the python and yaml style check by passing the SHELLCHECK_PATH as first argument to the check script

File: tools/check-shellcheck (hunk 2 above)
Comment (okurz) [reply tools/check-shellcheck]: Actually I first tried to get the quoting right in CMake but quicky gave up after my nightmares from 15 years ago about CMake shell script quoting bubbled up again :D So now I am calling this script from cmake

File: tools/check-yaml-syntax (hunk 3 above)
Comment (okurz) [reply tools/check-yaml-syntax]: yes, of course. I fixed that and changed accordingly same as I did for the python style check that was added in the meantime

**Review by Martchus — APPROVED** [review]

**Pull request merged**